src/test/regress/expected/create_role_propagation.out

```diff
@@ -228,22 +228,41 @@ SET ROLE non_dist_role_1;
 GRANT dist_role_1 TO dist_role_2;
 RESET ROLE;
 SET citus.enable_create_role_propagation TO ON;
-GRANT dist_role_3 TO non_dist_role_3;
+create role test_admin_role;
+grant dist_role_3 to test_admin_role with admin option;
```

Comment:
Can we change this into this so that we can actually test whether we propagate grantor. The latter grant stmt doesn't test this because of the none distributed role.

```suggestion
grant dist_role_3 to test_admin_role with admin option;
```

Reply:
I added test_admin_role just for granted by. I added a distributed role grant on next step. Is it sufficient?

Reply:
Regarding this GRANT stmt:
```sql
GRANT dist_role_3 TO dist_role_4 granted by test_admin_role;
```

We get different outputs on different nodes:
```sql
"member":"dist_role_4","role":"dist_role_3","grantor":"test_admin_role","admin_option":false}
...
"member":"dist_role_4","role":"dist_role_3","grantor":"postgres","admin_option":false}
...
"member":"dist_role_4","role":"dist_role_3","grantor":"postgres","admin_option":false}
```

So I got confused a bit, is this really expected?

Reply:
This is the bug related to metadata_sync grantor issue. I will fix it in another issue

Reply:
You're right . grantor was being set to NULL